### PR TITLE
Migrate from `npm` to `pnpm`

### PR DESCRIPTION
**Pull request by yeongsheng-tan**
Motivation: [Why you should prefer using pnpm over npm and yarn](https://refine.dev/blog/pnpm-vs-npm-and-yarn/#introduction)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,15 +3,20 @@
...
-    "integration-test": "cd integration-tests && npm run test-ci",
-    "lint-test": "cd integration-tests && npm run lint",
-    "release": "npm run copy-readme && cd project && npm run release",
+    "preinstall": "npx only-allow pnpm",
+    "lint-project": "pnpm -C project lint",
+    "ci-project": "pnpm -C project build",
+    "package": "pnpm -C project package",
+    "unit-test": "pnpm -C project test-ci",
+    "ci-integration-tests": "pnpm package && pnpm -C integration-tests add -D 'file://../project/anev-ts-mountebank-1.8.2.tgz' && pnpm -C integration-tests build",
+    "integration-test": "pnpm package && pnpm -C integration-tests add -D 'file://../project/anev-ts-mountebank-1.8.2.tgz' && pnpm -C integration-tests test-ci",
```
Comment (AngelaE) [inline package.json:17]: This does not work locally for me anymore, unless I start MB manually.
How do you run them locally?

What do you think about this:
1. rename `ci-integration-tests` => `build:integration-tests` 
2. `integration-test` => rename to `integration-tests-ci`
3. new script for running local similar to the 'integration-test-ci` but that does not start MB.

**Comment by AngelaE** [comment]
Thanks for this! Do you plan to push more changes? I do not mind if you rebase your changes on the base branch and force push to tidy things up.
I am really busy at the moment, that's why I am slow with the review.

It definitely makes sense to change to pnpm, but when I tried everything locally the integration tests failed because MB seems to be started after the tests. (I am on windows). Would be nice to fix that, but I prefer running MB extra anyway working locally - otherwise we would need to tear it down after each test run. Sometimes I like to look at the recorded requests etc.

**Comment by yeongsheng-tan** [comment]
> Thanks for this! Do you plan to push more changes? I do not mind if you rebase your changes on the base branch and force push to tidy things up. I am really busy at the moment, that's why I am slow with the review.
> 
> It definitely makes sense to change to pnpm, but when I tried everything locally the integration tests failed because MB seems to be started after the tests. (I am on windows). Would be nice to fix that, but I prefer running MB extra anyway working locally - otherwise we would need to tear it down after each test run. Sometimes I like to look at the recorded requests etc.

Hi @AngelaE thanks for the feedback. I have perhaps 1 more change to upgrade `chai` to `4.4.0`.

Noted to rebase my changes on the base branch and force push.

Let me try another way to fix `mb` starting slower than when `integration-test` kicks in.

File: package.json (hunk 1 above)
Comment (yeongsheng-tan) [reply package.json]: Applied the alias name changes and new local `integration-test-local` w/o starting `mb`. See https://github.com/AngelaE/ts-mountebank/pull/27/commits/ebe4601373c1568996fbdf41098e181960d0396a

**Comment by yeongsheng-tan** [comment]
If the above are alright and fixes your local integration tests, then I'll be happy to squash the commits and do a force push to clean up the history. Thanks @AngelaE

**Comment by yeongsheng-tan** [comment]
> If the above are alright and fixes your local integration tests, then I'll be happy to squash the commits and do a force push to clean up the history. Thanks @AngelaE

I've squahed as much of the commits as I deemed makes sense and forced pushed to this PR branch.

Very much appreciate your review and merge soon. Thank you. 😉

**Comment by yeongsheng-tan** [comment]
@AngelaE any chance you would like to have this reveiwed and merged? Wonder if there is still any motivation to move this along.

**Comment by AngelaE** [comment]
> @AngelaE any chance you would like to have this reveiwed and merged? Wonder if there is still any motivation to move this along.

I will try to merge it this weekend, apologies for not coming back to it. My life has been a bit mad.
Am definitely interested in changing to pnpm!

**Comment by AngelaE** [comment]
Thanks, everything is running much faster now! Definitely notice this in the gh actions!
Apologies for taking so long! I found the scripts confusing, which is why I put it aside around xmas. And then life became very stressful and I forgot 🙏 

Waiting for the MB process during the test run is neat and works very well (now that I started the correct script)!

I re-arranged some scripts now and added a commit hook to fail if the project is still referenced in the integration-tests, see https://github.com/AngelaE/ts-mountebank/pull/29 if you are interested. Nothing big, mostly re-names and I pushed some responsibilities down to the integration-test project.

**Comment by yeongsheng-tan** [comment]
> Thanks, everything is running much faster now! Definitely notice this in the gh actions! Apologies for taking so long! I found the scripts confusing, which is why I put it aside around xmas. And then life became very stressful and I forgot 🙏
> 
> Waiting for the MB process during the test run is neat and works very well (now that I started the correct script)!
> 
> I re-arranged some scripts now and added a commit hook to fail if the project is still referenced in the integration-tests, see #29 if you are interested. Nothing big, mostly re-names and I pushed some responsibilities down to the integration-test project.

Thank you so very much. Nice to see the tidy-up and renaming of package scripts.

**Pull request merged**